cardano-testnet/test/Test/Cli/Babbage/LeadershipSchedule.hs

```diff
@@ -0,0 +1,194 @@
+  conf@Conf { tempBaseAbsPath, tempAbsPath } <- H.noteShowM $
+    mkConf (ProjectBase base) (YamlFilePath configurationTemplate) tempAbsBasePath' Nothing
+
+  work <- H.note $ tempAbsPath </> "work"
+  H.createDirectoryIfMissing work
+
+  let
+    testnetOptions = BabbageOnlyTestnetOptions $ babbageDefaultTestnetOptions
+      { babbageNodeLoggingFormat = NodeLoggingFormatAsJson
+      }
+  tr@TestnetRuntime
+    { testnetMagic
+    , poolNodes
+    -- , wallets
+    -- , delegators
+    } <- testnet testnetOptions conf
+
+  poolNode1 <- H.headM poolNodes
+
+  env <- H.evalIO getEnvironment
+
+  poolSprocket1 <- H.noteShow $ nodeSprocket $ poolRuntime poolNode1
+
+  execConfig <- H.noteShow H.ExecConfig
+    { H.execConfigEnv = Last $ Just $
+      [ ("CARDANO_NODE_SOCKET_PATH", IO.sprocketArgumentName poolSprocket1)
+      ]
+      -- The environment must be passed onto child process on Windows in order to
+      -- successfully start that process.
+      <> env
+    , H.execConfigCwd = Last $ Just tempBaseAbsPath
+    }
+
+  tipDeadline <- H.noteShowM $ DTC.addUTCTime 210 <$> H.noteShowIO DTC.getCurrentTime
+
+  H.byDeadlineM 10 tipDeadline "Wait for two epochs" $ do
+    void $ execCli' execConfig
+      [ "query", "tip"
+      , "--testnet-magic", show @Int testnetMagic
+      , "--out-file", work </> "current-tip.json"
+      ]
+
+    tipJson <- H.leftFailM . H.readJsonFile $ work </> "current-tip.json"
+    tip <- H.noteShowM $ H.jsonErrorFail $ J.fromJSON @QueryTipLocalStateOutput tipJson
+
+    currEpoch <- case mEpoch tip of
+      Nothing -> H.failMessage callStack "cardano-cli query tip returned Nothing for EpochNo"
+      Just currEpoch -> return currEpoch
+
+    H.note_ $ "Current Epoch: " <> show currEpoch
+    H.assert $ currEpoch > 2
+
+  stakePoolId <- filter ( /= '\n') <$> execCli
+    [ "stake-pool", "id"
+    , "--cold-verification-key-file", poolNodeKeysColdVkey $ poolKeys poolNode1
+    ]
+
+  let poolVrfSkey = poolNodeKeysVrfSkey $ poolKeys poolNode1
+
+  id do
+    scheduleFile <- H.noteTempFile tempAbsPath "schedule.log"
+
+    leadershipScheduleDeadline <- H.noteShowM $ DTC.addUTCTime 180 <$> H.noteShowIO DTC.getCurrentTime
+
+    H.byDeadlineM 5 leadershipScheduleDeadline "Failed to query for leadership schedule" $ do
+      void $ execCli' execConfig
+        [ "query", "leadership-schedule"
+        , "--testnet-magic", show @Int testnetMagic
+        , "--genesis", shelleyGenesisFile tr
+        , "--stake-pool-id", stakePoolId
+        , "--vrf-signing-key-file", poolVrfSkey
+        , "--out-file", scheduleFile
+        , "--current"
+        ]
+
+    scheduleJson <- H.leftFailM $ H.readJsonFile scheduleFile
+
+    expectedLeadershipSlotNumbers <- H.noteShowM $ fmap (fmap slotNumber) $ H.leftFail $ J.parseEither (J.parseJSON @[LeadershipSlot]) scheduleJson
+
+    maxSlotExpected <- H.noteShow $ maximum expectedLeadershipSlotNumbers
```

Comment:
`maximum` is a partial pure function.  This code is exception-safe because we use `noteShow`.